### PR TITLE
devDeps: @lavamoat/allow-scripts@1.0.6->2.3.1

### PR DESCRIPTION
**Pull request by legobeat**
#6317

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -496,30 +496,54 @@
...
-      "usb": false,
-      "@trufflesuite/bigint-buffer": false,
-      "leveldown": false,
-      "puppeteer": false
+      "@eth-optimism/contracts>@ethersproject/hardware-wallets>@ledgerhq/hw-transport-node-hid>@ledgerhq/hw-transport-node-hid-noevents>node-hid": false,
+      "@eth-optimism/contracts>@ethersproject/hardware-wallets>@ledgerhq/hw-transport-node-hid>node-hid": false,
+      "@eth-optimism/contracts>@ethersproject/hardware-wallets>@ledgerhq/hw-transport-node-hid>usb": false,
+      "@ethereumjs/tx>ethereumjs-util>ethereum-cryptography>keccak": true,
+      "@ethereumjs/tx>ethereumjs-util>ethereum-cryptography>secp256k1": true,
+      "@sentry/react-native>@sentry/cli": false,
```
[the remark this answers is not in the document]
Comment (leotm) [reply package.json:509]: also noted `sentry-cli` has [these](https://github.com/getsentry/sentry-cli/blob/master/package.json#L32-L43) scripts

so with `false` we're currently disabling a [couple](https://github.com/getsentry/sentry-cli/blob/master/scripts/install.js#L334-L335) behaviours

@legobeat i think it'd be worth keeping us on `true`

```suggestion
      "@sentry/react-native>@sentry/cli": true,
```

so [`SENTRYCLI_USE_LOCAL`](https://github.com/getsentry/sentry-cli/blob/master/README.md?plain=1#L84)  works as intended (when [`downloadBinary`](https://github.com/getsentry/sentry-cli/blob/master/scripts/install.js#L189-L208) is called)

and so [checkVersion](https://github.com/getsentry/sentry-cli/blob/master/scripts/install.js#L303-L310) keeps us on the expected (compat) `sentry-cli` version

(i don't see sourcemaps affected which was my main concern, so our android/ios artifacts `~/index.android.bundle.map` and ``~/main.jsbundle.map`` should continue to get uploaded per release bundle)

**Review by leotm — CHANGES_REQUESTED** [review]
👌 only one suggestion RE Sentry

---

nb: Storybook testing depends on
- https://github.com/MetaMask/metamask-mobile/issues/6516

where we'll enable lifecycle scripts if/as needed

**Comment by leotm** [comment]
> ```
> @SocketSecurity ignore @lavamoat/allow-scripts@2.3.1
> @SocketSecurity ignore json-parse-even-better-errors@3.0.0
> @SocketSecurity ignore unique-slug@3.0.0
> @SocketSecurity ignore @npmcli/node-gyp@3.0.0
> @SocketSecurity ignore read-cmd-shim@4.0.0
> @SocketSecurity ignore @npmcli/run-script@6.0.1
> @SocketSecurity ignore npm-normalize-package-bin@3.0.1
> @SocketSecurity ignore write-file-atomic@5.0.1
> @SocketSecurity ignore @npmcli/fs@2.1.2
> @SocketSecurity ignore nopt@6.0.0
> @SocketSecurity ignore smart-buffer@4.2.0
> @SocketSecurity ignore @tootallnate/once@2.0.0
> @SocketSecurity ignore minipass-sized@1.0.3
> @SocketSecurity ignore err-code@2.0.3
> @SocketSecurity ignore promise-retry@2.0.1
> @SocketSecurity ignore minipass-flush@1.0.5
> @SocketSecurity ignore humanize-ms@1.2.1
> @SocketSecurity ignore @gar/promisify@1.1.3
> @SocketSecurity ignore minipass-pipeline@1.2.4
> @SocketSecurity ignore promise-inflight@1.0.1
> @SocketSecurity ignore is-lambda@1.0.1
> @SocketSecurity ignore make-fetch-happen@10.2.1
> @SocketSecurity ignore minipass-fetch@2.1.2
> @SocketSecurity ignore cacache@16.1.3
> @SocketSecurity ignore bin-links@4.0.1
> @SocketSecurity ignore @lavamoat/aa@3.1.2
> ```

nit: do you have a link to the prev report? curious to see the Socket Security issues on these deps

**Comment by leotm** [comment]
new `yarn.lock` conflict remaining then good 2 go!

**Review by leotm — APPROVED** [review]

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gudahtt) [reply package.json:516]: Looks like this PR landed with the Sentry CLI postinstall script still set to `false`, which has been causing Bitrise build failures

[the remark this answers is not in the document]
Comment (Gudahtt) [reply package.json:516]: It has been re-enabled in this PR: https://github.com/MetaMask/metamask-mobile/pull/7211

[the remark this answers is not in the document]
Comment (leotm) [reply package.json:516]: doh 🤦 thank you for catching this

**Pull request merged**